app/components/address_component.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class AddressComponent < ViewComponent::Base
+  def initialize(address:, raw_location: nil)
```

Comment:
i think we should possibly change the function siganature so it just takes an address? im not sure what this raw_location thing is doing but its on the Address record anyway

Reply:
I changed the call to this https://github.com/geeksforsocialchange/PlaceCal/blob/2270-migrate-address-component-to-view-component/app/views/events/show.html.erb#L32

and the original (on the `main` branch) is this https://github.com/geeksforsocialchange/PlaceCal/blob/main/app/views/events/show.html.erb#L32-L36

The commit adding `raw_location` to the signature is [b6959cf](https://github.com/geeksforsocialchange/PlaceCal/commit/b6959cf2950647c5f40abca9a703f501316f7429) and from looking around `raw_location_source` is the renamed `location` field on the `Event` model https://github.com/geeksforsocialchange/PlaceCal/blob/main/db/migrate/20181121122228_rename_events_location_column.rb#L5

Going by the history of that migration, [5bab05d](https://github.com/geeksforsocialchange/PlaceCal/commit/5bab05df938cf7cee9c86511803b97392c5bab9d#diff-3bd8a9f39831c937db535d4d188966a521cab79270c5b040df0cb450e342c59aR44) and searching in the code, it seems `app/models/events/base.rb` has become [`app/jobs/calendar_importer/events/base.rb`](https://github.com/geeksforsocialchange/PlaceCal/blob/main/app/jobs/calendar_importer/events/base.rb#L88) which is the only place `raw_location_from_source` is used.

Long story short :sweat_smile: I would keep the function signature the same, unless you feel up to further digging and manual testing. This would go back to the scope of the ticket: we can make a `TODO` to harmonize the usage of this field once the migration is done or attempt it within this PR. Let me know what you think :relaxed:

Reply:
ok agreed, and also i think with hindsight this is way too in the weeds to get into if we want to whizz through this directory, more just limit testing here :)

Reply:
Removed my original comment since it seemed to be duplicated across the review thread but apparently GitHub's review comment UI is misleading in its design. Anyway, what I recall writing was as follows:

The function signature `initialize(address:, raw_location: nil)` was added in commit [b6959cf](https://github.com/geeksforsocialchange/PlaceCal/commit/b6959cf2950647c5f40abca9a703f501316f7429) for displaying events without an address.

This commit [5bab05d](https://github.com/geeksforsocialchange/PlaceCal/commit/5bab05df938cf7cee9c86511803b97392c5bab9d) shows that the `raw_location_from_source` was a renaming of the `location` field on the `Event` model

The `raw_location` field isn't required for showing partners' info since they will most likely have filled in an address as part of onboarding.